jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/SecurityContactsCollector.java

```diff
@@ -53,6 +52,47 @@ public List<SecurityContactsVH> fetchSecurityContactsInfo(SubscriptionVH subscri
         LOGGER.info("Target Type : {}  Total: {} ", "Batch Account", securityContactsList.size());
         return securityContactsList;
     }
+
+    private List<AutoProvisioningSettingsVH>fetchAutoProvisioningSettingsList(SubscriptionVH subscription){
+        List<AutoProvisioningSettingsVH>autoProvisioningSettingsVHList=new ArrayList<>();
+
+        try {
+            String apiUrlTemplate="https://management.azure.com/%s/providers/Microsoft.Security/autoProvisioningSettings?api-version=2017-08-01-preview";
+            String accessToken = azureCredentialProvider.getToken(subscription.getTenant());
+            String url = String.format(apiUrlTemplate, URLEncoder.encode("/subscriptions/"+subscription.getSubscriptionId(),java.nio.charset.StandardCharsets.UTF_8.toString()));
+            LOGGER.info("The url is {}",url);
+
+            String response = CommonUtils.doHttpGet(url, "Bearer", accessToken);
+            LOGGER.info("Response is :{}",response);
+            JsonObject responseObj = new JsonParser().parse(response).getAsJsonObject();
+            JsonArray autoProvisioningObjects = responseObj.getAsJsonArray("value");
+
+            for(JsonElement autoProvisioningElement:autoProvisioningObjects){
+                AutoProvisioningSettingsVH autoProvisioningSettingsVH=new AutoProvisioningSettingsVH();
+                JsonObject autoProvisioningObject=autoProvisioningElement.getAsJsonObject();
+                String id=autoProvisioningObject.get("id").getAsString();
```

Comment:
Lets use string literals for reading the params- id,name,type etc

Reply:
Done